packages/salesforcedx-utils-vscode/src/cli/commandExecutor.ts

```diff
@@ -155,28 +156,36 @@ export class CompositeCliCommandExecution implements CommandExecution {
         }
       });
     }
-    this.processErrorSubject.subscribe(() => {
+    this.processErrorSubject.subscribe(e => {
       if (timerSubscriber) {
         timerSubscriber.unsubscribe();
       }
+      CommandEventStream.getInstance().post({ type: CommandEventType.ERROR, error: `${e}` });
     });
 
-    this.processExitSubject.subscribe(() => {
+    this.processExitSubject.subscribe(exitCode => {
       if (timerSubscriber) {
         timerSubscriber.unsubscribe();
       }
+      if (exitCode !== undefined) {
+        CommandEventStream.getInstance().post({ type: CommandEventType.EXIT_CODE, exitCode });
+      }
     });
   }
 
   public successfulExit() {
+    CommandEventStream.getInstance().post({ type: CommandEventType.EXIT_CODE, exitCode: 0 });
     this.exitSubject.next(0);
   }
 
   public failureExit(e?: {}) {
     if (e) {
+      // eslint-disable-next-line @typescript-eslint/no-base-to-string, @typescript-eslint/restrict-template-expressions
+      CommandEventStream.getInstance().post({ type: CommandEventType.ERROR, error: `${e}` });
```

Comment:
Is there a safer way to do this without all the eslint bypasses?

Reply:
I'll see what I can do.